text/0015-gas-payments.md

```diff
@@ -0,0 +1,90 @@
+# Casper Payments: Gas Payment Concepts
+
+## High level view
+This is the first of a few CEPs on Casper Payment model. 
+The motivation for this work is to make Casper much more user and dev 
+friendly than it is now. There are a few topics here, that mix. The change 
+to one of them might cause the domino effect on others, but we'll 
+try to chunk them into logical units.
+
+#### Token Standard 
+Most of the systems built on top of smart contracting platforms use tokens 
+and it is reasonable to put a lot of effort in designing a good token standard 
+or discover gaps in core functionalities, that should be fixed before 
+the Mainnet. The most popular token standard right now is the ERC20. We have 
+implemented its Casper version here https://github.com/casperlabs/erc20.
+We think we can do better and offer our future community a token standard 
+that avoids the "approve and call" pitfall. 
+This work will be the subject of future CEP.
+
+#### CSPR Interface
+One of the most important lessons we (as a blockchain community) learned 
+from the Ethereum is that it's problematic when the platform's native token 
+doesn't implement a good token standard. After our Token Standard is defined 
+we will consider CSPR reimplementation to support the Standard.
+This work will be the subject of future CEP.
+
+#### Gas Payments
+This part covers the topic of how users pay for the gas. It doesn't cover 
+gas model itself. This work is a subject for current CEP.
+
+#### Smart contract and accounts unification
+The new model of our accounting/operational unit that should easily allow for 
+all above. That means merging accounts and contracts (and urefs?) addressing 
+spaces, enabling easy transfer of tokens between accounts and addresses, 
+making uref sharing easy and ways of calling contracts.
+This work will be the subject of future CEP.
+
+## Gas Payments
+In this section, we'll cover ways of paying for deploys that were discussed
+in a past and we'll decide what we want to support.
+
+### Self Payment
+This is the simplest case. User should be able to use its own CSPRs to pay
+for the execution of its own deploy (WASM or contract call).
+
+### Delegeted Payment
+Alice and Bob should be able to execute the following scenario:
+
+1. Alice creates a Session Code (contract call) and signs it with her private key. 
+2. Alice sends it to Bob.
+3. Bob adds Payment Code information and signs it with his private key.
+4. Bob sends it to the network.
```

Comment:
This could work in two ways:
1. (proposed version), where Alice creates a session code with signatures and sends it to Bob. Then Bob adds payment and of course could send it back to Alice.
2. Bob creates a session code (sets contract hash + arguments + session_from), signes it a payer and sends to Alice for the final session code signature. Then it's up to Alice to send it to the network or back to Bob.

Our of those two, the first one I expect the first one to be used much more often, as it should be in the Bob's best interest to execute the transaction. On the other hand Bob would be in the best position to front-run Alice. What do you think?